src/test/java/se/sundsvall/billingdatacollector/service/CollectorServiceTest.java

```diff
@@ -0,0 +1,62 @@
+package se.sundsvall.billingdatacollector.service;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import se.sundsvall.billingdatacollector.TestDataFactory;
+import se.sundsvall.billingdatacollector.integration.billingpreprocessor.BillingPreprocessorIntegration;
+import se.sundsvall.billingdatacollector.integration.opene.OpenEIntegration;
+import se.sundsvall.billingdatacollector.model.BillingRecordWrapper;
+import se.sundsvall.billingdatacollector.service.mapper.BillingRecordDecorator;
+
+@ExtendWith(MockitoExtension.class)
+class CollectorServiceTest {
+
+	@Mock
+	private OpenEIntegration mockOpenEIntegration;
+
+	@Mock
+	private BillingPreprocessorIntegration mockBillingPreprocessorIntegration;
+
+	@Mock
+	private BillingRecordDecorator mockDecorator;
+
+	private static final String FAMILY_ID = "123";
+
+	private CollectorService collectorService;
+
+	@BeforeEach
+	void setUp() {
+		when(mockDecorator.getSupportedFamilyId()).thenReturn(FAMILY_ID);
+		collectorService = new CollectorService(mockOpenEIntegration, mockBillingPreprocessorIntegration, List.of(mockDecorator));
```

Comment:
Add `@InjectMocks` on `private CollectorService collectorService` instead?

Reply:
Unfortunately I don't think it will work in this case as the `List.of(mockDecorator)` is added to the `Map` in the constructor.

Reply:
ok